autoreadme.go

```diff
@@ -207,6 +217,53 @@ func getDoc(dir string) (Doc, error) {
 	}, nil
 }
 
+func renderExamples(bi *build.Package) (map[string]string, error) {
+	filter := func(fi os.FileInfo) bool {
+		if fi.IsDir() {
+			return false
+		}
+		nm := fi.Name()
+		for _, f := range bi.XTestGoFiles {
```

Comment:
These slices are sorted and if you're parsing the tests then the xtests a duplicate example will always be overridden by the one in xtests (assuming everything compiles). Perfectly deterministic and prefers xtests.